src/avro_util.erl

```diff
@@ -62,6 +63,72 @@
 -type validate_fun() :: fun((type_or_name(), validate_hist()) -> ok).
 
 %%%_* APIs =====================================================================
+%% @doc Checks whether two schemas are compatible in sense that an event
+%% encoded with the writer schema can be decoded by the reader schema.
+%% Schema compatibility is defined in
+%% https://avro.apache.org/docs/1.8.1/spec.html#Schema+Resolution
+%% @end
+-spec avro_schema_compatible(avro_type(), avro_type()) ->
+                                true | false.
+avro_schema_compatible(Reader, Writer)
+  when ?IS_RECORD_TYPE(Reader) andalso ?IS_RECORD_TYPE(Writer) ->
+  SameFullName =
```

Comment:
`avro:is_same_type/2` checks full-name equality.
since we call `is_same_type` in all other clauses,
it's better to have a consistent style here.